### PR TITLE
Release/v0.31.4

### PR DESCRIPTION
**Pull request by Liamsi**
https://github.com/tendermint/tendermint/blob/b6da8880c22202a7061ae2093ef01819438a136e/CHANGELOG.md

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,42 @@
 # Changelog
 
+## v0.31.4
+
+*April 12th, 2019*
+
+This release fixes a regression from v0.31.3 which used the peer's `SocketAddr` to add the peer to
```
Comment (melekes) [inline CHANGELOG.md]: 😞

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,42 @@
...
+
+### BREAKING CHANGES:
+
+* Go API
+  - [p2p] [\#3545](https://github.com/tendermint/tendermint/pull/3545) The `AddrBook` interface method `MarkAsGood` now only takes a `p2p.ID` instead of a `p2p.NetAddress`
+  - [p2p] [\#3011](https://github.com/tendermint/tendermint/pull/3011) Remove `ListOfKnownAddresses` from the `AddrBook` interface
+
+### IMPROVEMENTS:
+- [p2p] [\#3463](https://github.com/tendermint/tendermint/pull/3463) Do not log "Can't add peer's address to addrbook" error for a private peer
+- [p2p] [\#3547](https://github.com/tendermint/tendermint/pull/3547) Fix a couple of annoying typos (@mdyring)
```
Comment (melekes) [inline CHANGELOG.md]: should we even include such minor things?

Comment (Liamsi) [reply CHANGELOG.md]: I'm not entirely sure where to draw the line here. At least for external contributors, I think we should include their contributions even if minor.

File: CHANGELOG.md (hunk 1 above)
Comment (Liamsi) [reply CHANGELOG.md]: Do you want me to change sth here? Or is this 😞just an expression of sadness about that regression? :-D

Again, this is probably sth we could have catched by running more extensive (actual) autm. testing/testnet, or, what @greg-szabo and @mircea-c call automated QA.

Comment (melekes) [reply CHANGELOG.md]: > Do you want me to change sth here? 
no

Comment (xla) [reply CHANGELOG.md]: Can we introduce emoji prefixes for the different entries? :cry: for regressions for example. :p

File: CHANGELOG.md (hunk 2 above)
Comment (xla) [reply CHANGELOG.md]: If the changelog is a channel for communication with peers who contribute or use the codebase we should announce all the things, including ADRs, maybe it sparks more collaboration on those.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,42 @@
...
+As a side-effect it makes these RPC endpoint more difficult to DoS and fixes a performance degradation in `ExecCommitBlock`.
+
+Special thanks to external contributors on this release:
+@brapse, @guagualvcha, @mydring
+
+### BREAKING CHANGES:
+
+* Go API
+  - [p2p] [\#3545](https://github.com/tendermint/tendermint/pull/3545) The `AddrBook` interface method `MarkAsGood` now only takes a `p2p.ID` instead of a `p2p.NetAddress`
+  - [p2p] [\#3011](https://github.com/tendermint/tendermint/pull/3011) Remove `ListOfKnownAddresses` from the `AddrBook` interface
```
Comment (ebuchman) [inline CHANGELOG.md]: p2p isn't listed in https://github.com/tendermint/tendermint/#semantic-versioning so technically we don't need to document these :)

**Comment by Liamsi** [comment]
> Can we also add the note from https://github.com/tendermint/tendermint/releases/tag/v0.31.3 that we forgot to put in changelog for v0.31.3?

OK, will do.

File: CHANGELOG.md (hunk 2 above)
Comment (Liamsi) [reply CHANGELOG.md]: https://github.com/tendermint/tendermint/pull/3550

File: CHANGELOG.md (hunk 3 above)
Comment (Liamsi) [reply CHANGELOG.md]: removed in https://github.com/tendermint/tendermint/pull/3550

File: CHANGELOG.md (hunk 1 above)
Comment (ebuchman) [reply CHANGELOG.md]: Lol great idea!

**Pull request merged**